osu.Game/Screens/Edit/Compose/Components/SelectionScaleHandler.cs

```diff
@@ -32,8 +32,26 @@ public partial class SelectionScaleHandler : Component
         /// </remarks>
         public Bindable<bool> CanScaleDiagonally { get; private set; } = new BindableBool();
 
+        /// <summary>
+        /// Whether scaling anchored by the selection origin can currently be performed.
+        /// </summary>
+        public Bindable<bool> CanScaleSelectionOrigin { get; private set; } = new BindableBool();
+
+        /// <summary>
+        /// Whether scaling anchored by the center of the playfield can currently be performed.
+        /// </summary>
+        public Bindable<bool> CanScalePlayfieldOrigin { get; private set; } = new BindableBool();
+
         public Quad? OriginalSurroundingQuad { get; protected set; }
 
+        /// <summary>
+        /// Clamp scale where selection does not exceed playfield bounds or flip.
+        /// </summary>
+        /// <param name="origin">The origin from which the scale operation is performed</param>
+        /// <param name="scale">The scale to be clamped</param>
+        /// <returns>The clamped scale vector</returns>
+        public virtual Vector2 GetClampedScale(Vector2 scale, Vector2? origin = null) => scale;
```

Comment:
Same for this, I don't think this belongs in the base class. Should be moved to derived.